src/dmd/dstruct.d

```diff
@@ -297,13 +297,32 @@ extern (C++) class StructDeclaration : AggregateDeclaration
             return;
         }
 
-        // 0 sized struct's are set to 1 byte
         if (structsize == 0)
         {
             hasNoFields = true;
             alignsize = 1;
-            if (classKind != classKind.c) // C gets a struct size of 0
-                structsize = 1;
+
+            // A fine mess of what size a zero sized struct should be
+            final switch (classKind)
+            {
+                case ClassKind.d:
+                case ClassKind.cpp:
+                    structsize = 1;
+                    break;
+
+                case ClassKind.c:
+                case ClassKind.objc:
+                    if (target.c.bitFieldStyle == TargetC.BitFieldStyle.MS)
+                    {
+                        /* Undocumented MS behavior for:
+                         *   struct S { int :0; };
+                         */
+                        structsize = 4;
```

Comment:
What about `short :0`, or `long long :0`? Does the type dictate the padding after it?

Reply:
Surprisingly, no. It's always 4. Even `long long :0; long long :0;` comes out as 4. The type of the 0 length bit field has no effect.

Reply:
This peculiar behavior only happens when the struct size is zero.

Reply:
OK, just making sure that we aren't missing a test.